src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1433,28 +1435,45 @@ def create_function(resource_group_name, name, storage_account, plan=None,
     site_config = SiteConfig(app_settings=[])
     functionapp_def = Site(location=None, site_config=site_config)
     client = web_client_factory()
+
     if consumption_plan_location:
         locations = list_consumption_locations()
         location = next((l for l in locations if l['name'].lower() == consumption_plan_location.lower()), None)
         if location is None:
             raise CLIError("Location is invalid. Use: az functionapp list-consumption-locations")
         functionapp_def.location = consumption_plan_location
+        functionapp_def.kind = 'functionapp'
     else:
         if is_valid_resource_id(plan):
             plan = parse_resource_id(plan)['name']
         plan_info = client.app_service_plans.get(resource_group_name, plan)
         location = plan_info.location
+        is_linux = plan_info.reserved
+        if is_linux:
+            functionapp_def.kind = 'functionapp,linux'
+            site_config.app_settings.append(NameValuePair('FUNCTIONS_EXTENSION_VERSION', 'beta'))
+            site_config.app_settings.append(NameValuePair('MACHINEKEY_DecryptionKey',
+                                                          str(binascii.hexlify(os.urandom(32)).decode()).upper()))
```

Comment:
if the goal is just to generate a unique hex string, and there is no hard length requirement involved, maybe you can just go with `uuid.uuid4().hex`, which can by-pass the python 2 and 3 string/bytes difference the code is trying to handle here.

Reply:
it has to be at least 32 byte encryption key in hex format. I don't know if uuid can be used for crypto values, I just read the docs for [`os.urandom`](https://docs.python.org/3/library/os.html#os.urandom) and it seemed to do what I want. Anyway, this seems to work as I'd expect on 2.7 and 3.x, but thanks for the pointer, i'll look at uuid.